website/content/en/preview/upgrade/upgrade-ref.md

```diff
@@ -0,0 +1,606 @@
+
+
+**subnetSelector example (v1alpha)**
+
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+...
+spec:
+  subnetSelector:
+    karpenter.sh/discovery: karpenter-demo
+```
+
+**SubnetSelectorTerms.tags example (v1beta1)**
+```
+apiVersion: karpenter.k8s.aws/v1beta1
+kind: EC2NodeClass
+...
+spec:
+  subnetSelectorTerms:
+  - tags:
+      karpenter.sh/discovery: karpenter-demo
+```
+
+**subnetSelector example (v1alpha)**
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+...
+spec:
+  subnetSelector:
+    aws::ids: subnet-123,subnet-456
+```
+
+**subnetSelectorTerms.id example (v1beta1)**
+```
+apiVersion: karpenter.k8s.aws/v1beta1
+kind: EC2NodeClass
+...
+spec:
+  subnetSelectorTerms:
+  - id: subnet-123
+  - id: subnet-456
+```
+
+**securityGroupSelector example (v1alpha)**
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+...
+spec:
+  securityGroupSelector:
+    karpenter.sh/discovery: karpenter-demo
+```
+
+**securityGroupSelectorTerms.tags example (v1beta1)**
+```
+apiVersion: compute.k8s.aws/v1beta1
+kind: EC2NodeClass
+...
+spec:
+  securityGroupSelectorTerms:
+  - tags:
+      karpenter.sh/discovery: karpenter-demo
+```
+
+**securityGroupSelector example (v1alpha)**
+
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+...
+spec:
+  securityGroupSelector:
+    aws::ids: sg-123, sg-456
+```
+
+
+**securityGroupSelectorTerms.id example (v1beta1)**
+```
+apiVersion: compute.k8s.aws/v1beta1
+kind: EC2NodeClass
+...
+spec:
+  securityGroupSelectorTerms:
+  - id: sg-123
+  - id: sg-456
+```
+
+
+**amiSelector example (v1alpha)**
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+...
+spec:
+  amiSelector:
+    karpenter.sh/discovery: karpenter-demo
+```
+
+
+**amiSelectorTerms.tags example (v1beta1)**
+```
+apiVersion: compute.k8s.aws/v1beta1
+kind: EC2NodeClass
+...
+spec:
+  amiSelectorTerms:
+  - tags:
+      karpenter.sh/discovery: karpenter-demo
+```
+
+**amiSelector example (v1alpha)**
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+...
+spec:
+  amiSelector:
+    aws::ids: ami-123,ami-456
+```
+
+**amiSelectorTerms example (v1beta1)**
+```
+apiVersion: compute.k8s.aws/v1beta1
+kind: EC2NodeClass
+...
+spec:
+  amiSelectorTerms:
+  - id: ami-123
+  - id: ami-456
+```
+
+
+**amiSelector example (v1alpha)**
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+...
+spec:
+  amiSelector:
+    aws::name: my-name1,my-name2
+    aws::owners: 123456789,amazon
+```
+
+**amiSelectorTerms.name example (v1beta1)**
+```
+apiVersion: compute.k8s.aws/v1beta1
+kind: EC2NodeClass
+...
+spec:
+  amiSelectorTerms:
+  - name: my-name1
+    owner: 123456789
+  - name: my-name2
+    owner: 123456789
+  - name: my-name1
+    owner: amazon
+  - name: my-name2
+    owner: amazon
+```
+
+### LaunchTemplateName
+
+The `spec.launchTemplateName` field for referencing unmanaged launch templates within Karpenter has been removed. 
```

Comment:
Do we want to provide any more detail on this? Perhaps we could link to the design doc that proposes and confirms that we decided to drop this field.

Reply:
I added a link to the design doc: https://github.com/aws/karpenter/blob/main/designs/unmanaged-launch-template-removal.md